sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -731,6 +909,20 @@ case class Sample(
   }
 
   override protected def otherCopyArgs: Seq[AnyRef] = isTableSample :: Nil
+
+  override def sql: String = child match {
+    case SubqueryAlias(alias, _: NonSQLPlan) =>
+      val repeatable = if (withReplacement) s" REPEATABLE ($seed)" else ""
+      s"$alias TABLESAMPLE(${upperBound * 100} PERCENT)$repeatable"
+
+    case SubqueryAlias(alias, grandChild) =>
+      val repeatable = if (withReplacement) s" REPEATABLE ($seed)" else ""
+      s"${grandChild.sql} TABLESAMPLE(${upperBound * 100} PERCENT)$repeatable $alias"
+
+    case _ =>
+      val repeatable = if (withReplacement) s" REPEATABLE ($seed)" else ""
```

Comment:
`repeatable` repeated three times (and don't think it's on purpose despite the name :))